exemplars_parent/exemplars/src/main/java/io/prometheus/client/exemplars/api/Exemplar.java

```diff
@@ -0,0 +1,70 @@
+package io.prometheus.client.exemplars.api;
+
+/**
+ * Immutable data class holding an Exemplar.
+ */
+public class Exemplar {
```

Comment:
The `ExemplarSampler` implementations use the `Exemplar` data model, so if we move the model to simpleclient, we will need to move all `ExemplarSampler` implementations to simpleclient as well. Currently it's not much, but if in the future people provide more sophisticated implementations it might be worthwhile to have this in a separate module.

Reply:
My point is that the core data model should all live in one place, not spread across N modules that have to be pieced together by someone trying to work with this. It's kinda like having an entire module just to handle timestamps.

Personally I wouldn't have the core simpleclient depending on any other modules at all, rather make this something that the user has to add an extra module to enable, rather than silently creating the possibility for dependency hell on upgrade. simpleclient thus far has purposefully avoided any dependencies for that reason, it should always be safe to have simpleclient instrumentation in place.

This is also still not the data model of an OM exemplar, as timestamps are optional.